core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -260,6 +262,17 @@ public AbstractClientOptionsBuilder endpointRemapper(
         return option(ClientOptions.ENDPOINT_REMAPPER, endpointRemapper);
     }
 
+    /**
+     * Sets the {@link AutoCloseable} which will be called whenever this {@link RequestContext} is popped
```

Comment:
This does not set an `AutoCloseable` but set a `Supplier` which produces an `AutoCloseable`. Should we update the Javadoc to reflect the actual behavior correctly?